feat: add support for Deezer song and playlist links

mathisdev7 opened the pull request:
Add Deezer song link and playlist support

- Implement support for Deezer song extraction via song URLs.
- Add functionality to fetch and parse Deezer playlists.
- Enable fetching of Deezer playlist tracks and their metadata.
- Improve the extraction process for Deezer content.

mathisdev7 committed:
feat: add support for Deezer song and playlist links

mathisdev7 committed:
feat: add album support

retrouser955 commented:
@mathisdev7 Deezer actually has a [public API](https://developers.deezer.com/api) for these sorts of things which doesn't even need an API key. I recommend using scraping as a last resort rather than a first entry.

mathisdev7 commented:
Yes, but how can I parse the playlist/song from the link a user provided? Should I parse a link like this: https://dzr.page.link/[id], get the ID, and find the song/playlist using the API?

retrouser955 commented:
> Yes, but how can I parse the playlist/song from the link a user provided? Should I parse a link like this: [https://dzr.page.link/[id]](https://dzr.page.link/%5Bid%5D), get the ID, and find the song/playlist using the API?

You can just fetch the URL and get the redirect url. Then you can parse it.

mathisdev7 committed:
fix(api): update Deezer shortlink domain detection

Deezer changed its shortlink domain from 'deezer.page.link' to 'dzr.page.link'. This update ensures proper detection and parsing of user-provided links.

mathisdev7 commented:
Yeah, it seems I strayed too far from the original issue. Deezer changed their domain name from 'deezer.page.link' to 'dzr.page.link'.

mathisdev7 committed:
fix: revert to old parsing system to transition to the new one

mathisdev7 commented:
should be good now, lmk

retrouser955 reviewed (approved):
Should be good now :). LGTM!
